src/shared/resourcefetcher/httpResourceFetcher.ts

```diff
@@ -31,14 +31,12 @@ const patchedGot = got.extend({
 // VSC 1.44.2 seems to work, but on C9 it does not?
 const MIN_VERSION_FOR_GOT = '1.47.0'
 
-// Minimal interface for hooking into download + file write streams
-interface FetcherStreams {
+/** Promise that resolves/rejects when all streams close. Can also access streams directly. */
+type FetcherResult = Promise<void> & {
```

Comment:
What does the `Promise<void>` add here? Seems unconventional to have a Promise that also carries data with it immediately...is this more intuitive than the previous impl?

Reply:
Attaching data to a promise is a way to offer extensibility without obstructing the common case. I've been seeing a few JS libraries do this (like `got`).

Not sure if it's more intuitive or not. I thought that having to `await` a field would seem more confusing but I'm not sure. Could always just make a class, though would it be too heavy?